src/layer-events-hoc.tsx

```diff
@@ -132,57 +137,118 @@ function layerMouseTouchEvents(
       const { map } = this.context;
 
       evt.features.forEach((feature: Feature) => {
+        // Set hover
         const { id } = feature.properties;
         this.hover = [id];
-        if (this.areFeaturesDraggable(children)) {
-          map.dragPan.disable();
-          this.isDragging = true;
 
-          map.on('touchmove', this.onDragMove);
-          map.once('touchend', this.onDragUp.bind(this, 'touchmove'));
+        if (!this.areFeaturesDraggable(children)) {
+          return
         }
+
+        // User did this on a feature
+        map.dragPan.disable();
+
+        this.onFeatureDown('touchstart');
+
+        map.once('touchend', () => {
+          map.dragPan.enable()
+        });
       });
     };
 
+    public onFeatureDown = (startEvent: string) => {
+      const moveEvent = startEvent === 'mousedown' ? 'mousemove' : 'touchmove';
+      const endEvent = startEvent === 'mousedown' ? 'mouseup' : 'touchend';
+      const { map } = this.context;
+
+      this.isGrabbing = true;
+      this.hasDragged = false;
+
+      this.draggedChildren = [];
+      map.once(moveEvent, this.onFeatureDragStart);
+      map.on(moveEvent, this.onFeatureDrag);
+      map.once(endEvent, this.onFeatureDragEnd);
```

Comment:
Can you listen off those events `onFeatureDragEnd` / on `componentWillUnmount` ?